community/kernel/src/test/java/org/neo4j/kernel/impl/store/NeoStoresTest.java

```diff
@@ -756,9 +774,13 @@ public void shouldCloseAllTheStoreEvenIfExceptionsAreThrown() throws Exception
         {
             // then
             assertEquals( "Cannot stop in state: rotating", ex.getMessage() );
-            // and the page cache closes with no errors
-            pageCache.close();
         }
+
+        thread.interrupt();
+        thread.join();
```

Comment:
Could you use an single-threaded `ExecutorService` instead of a `Thread` directly? That way you can get the `RuntimeException` out of the `Future.get()` method. You can use `Future.cancel(true)` to interrupt the thread. (Or perhaps even make it interrupt itself? I don't know)